op-batcher/batcher/espresso/streamer.go

```diff
@@ -0,0 +1,232 @@
+package espresso
+
+import (
+	// #cgo darwin,arm64 LDFLAGS: -framework CoreFoundation -framework SystemConfiguration
+	"C"
+	"cmp"
+	"context"
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"slices"
+
+	espressoClient "github.com/EspressoSystems/espresso-network-go/client"
+	espressoLightClient "github.com/EspressoSystems/espresso-network-go/light-client"
+	espressoTypes "github.com/EspressoSystems/espresso-network-go/types"
+	espressoVerification "github.com/EspressoSystems/espresso-network-go/verification"
+
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+// espresso-network-go's HeaderInterface currently lacks a function to get this info,
+// although it is present in all header versions
+func getFinalizedL1(header *espressoTypes.HeaderImpl) *espressoTypes.L1BlockInfo {
```

Comment:
This is the most recent L1 finalized according to the Espresso network right? And it might be different from the most recent L1 finalized block according to the L1 client used by the batcher / op node for example?